admin/sql/updates/20200630-mbs-10921-unlist-unrelated-edits.sql

```diff
@@ -0,0 +1,286 @@
+)
+RETURNING *;
+
+\echo MBS-10921 (2/12): Unlist “Edit relationship” edits from artists’ history...
+DELETE FROM edit_artist
+WHERE (edit, artist) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 91
+    AND jsonb_extract_path_text(ed.data, 'type0') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'type1') = 'work'
+    AND jsonb_extract_path_text(ed.data, 'link', 'entity0', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(ed.data, 'link', 'entity0', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 900 -- of 86841
+)
+RETURNING *;
+
+\echo MBS-10921 (3/12): Unlist “Remove relationship” edits from artists’ history...
+DELETE FROM edit_artist
+WHERE (edit, artist) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 92
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'link', 'type', 'entity0_type') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'link', 'type', 'entity1_type') = 'work'
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'entity0', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(ed.data, 'relationship', 'entity0', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 50 -- of 1799
+)
+RETURNING *;
+
+-- II) Recordings
+
+\echo MBS-10921 (4/12): Unlist “Reorder relationships” edits from artists’ history...
+DELETE FROM edit_recording
+WHERE (edit, recording) IN (
+    SELECT ea.edit, ea.artist
+    FROM edit_artist ea
+    JOIN edit e ON ea.edit = e.id
+    JOIN edit_data ed ON e.id = ed.edit
+    WHERE e.id >= 70658947 AND e.id < 71006471
+    AND e.type = 99
+    AND jsonb_extract_path_text(ed.data, 'link_type', 'entity0_type') = 'recording'
+    AND jsonb_extract_path_text(ed.data, 'link_type', 'entity1_type') = 'work'
+    AND jsonb_extract_path_text(data, 'relationship_order', '0', 'relationship', 'entity1', 'id')::int IN (
+        SELECT id FROM recording
+    )
+    AND jsonb_extract_path_text(data, 'relationship_order', '0', 'relationship', 'entity1', 'id')::int NOT IN (
+        SELECT r.id
+        FROM recording r
+        JOIN artist_credit ac ON r.artist_credit = ac.id
+        JOIN artist_credit_name acn ON ac.id = acn.artist_credit
+        WHERE acn.artist = ea.artist
+    )
+    LIMIT 50 -- of 534
```

Comment:
That gets the recording (actually the work) at index 0, but `relationship_order` is a list of relationships:
https://github.com/metabrainz/musicbrainz-server/blob/4d3b7779ffd1d6753cd784fce21cd6f5b0cd3e8b/lib/MusicBrainz/Server/Edit/Relationship/Reorder.pm#L158-L159

`entity1` here gets the work, not the recording since type1 = work.

```suggestion
    AND EXISTS (
        SELECT 1
          FROM recording r
          JOIN (SELECT jsonb_array_elements(ed.data->'relationship_order') AS item) x
            ON (x.item#>'{relationship,entity0,id}')::INT = r.id
    )
    AND ea.artist NOT IN (
        SELECT acn.artist
          FROM artist_credit_name acn
          JOIN recording r ON r.artist_credit = acn.artist_credit
          JOIN (SELECT jsonb_array_elements(ed.data->'relationship_order') AS item) x
            ON (x.item#>'{relationship,entity0,id}')::INT = r.id
    )
    LIMIT 50 -- of 534
```

Reply:
Since it is just reordering recording-work relationships, I assume that every relationship holds the same `entity0.id`.
So I’m not sure it is necessary to run a more complex query. I will check edit data to confirm that.

Reply:
Oh ok, your analysis is probably correct here

Reply:
Fixed `s/entity1/entity0/` in https://github.com/metabrainz/musicbrainz-server/pull/1581/commits/ea6b10dd7562bed837af8b73ae7a49705bc12337.